File: app/src/main/java/com/giulliano/mercado/viewModel/MainViewModel.java
package com.giulliano.mercado.viewModel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import com.giulliano.mercado.model.Item;
import com.giulliano.mercado.repository.ItemRepository;

import java.util.List;

/**
 * Created by Giulliano on 06/03/2018.
 */

public class MainViewModel extends AndroidViewModel {

    private ItemRepository mRepository;

    private LiveData<List<Item>> mAllItens;

    public MainViewModel (Application application) {
        super(application);
        mRepository = new ItemRepository(application);
        mAllItens = mRepository.getAllItens();
    }

    LiveData<List<Item>> getAllWords() { return mAllItens; }

    public void insert(Item word) { mRepository.insert(word); }
}

File: app/src/main/java/com/giulliano/mercado/viewModel/MainViewModel.kt
package com.giulliano.mercado.viewModel

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import android.arch.lifecycle.ViewModel
import com.giulliano.mercado.model.Item
import com.giulliano.mercado.repository.ItemRepository
import javax.inject.Inject




/**
 * Created by Giulliano on 28/02/2018.
 */
class MainViewModel : ViewModel() {

    private var itens: LiveData<List<Item>>? = null
    private var itemRepo: ItemRepository? = null

    fun init() {
        if (this.itens != null) {
            return
        }
        itens = itemRepo?.items
    }


    @Inject
    fun MainViewModel(userRepo: ItemRepository) {
        this.itemRepo = userRepo
    }

    fun getItens(): LiveData<List<Item>>? {
        return this.itens
    }


}